core/chains/evm/logpoller/parser.go

```
@@ -0,0 +1,493 @@
+package logpoller
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+)
+
+const (
+	sequenceFieldAlias = "sequence"
+	sequenceField      = "CONCAT(blockHash, '-', txHash, '-', logIndex)"
+	blockFieldName     = "block_number"
+	timestampFieldName = "timestamp"
+	txHashFieldName    = "tx_hash"
+	eventSigFieldName  = "event_sig"
+	chainIDFieldName   = "evm_chain_id"
+	evmLogsTableName   = "evm.logs"
+)
+
+type dslQueryProps struct {
+	ChainID *big.Int
+	Filter  query.KeyFilter
+	Limiter query.LimitAndSort
+}
+
+func (p *dslQueryProps) hasCursorLimit() bool {
+	return p.Limiter.Limit.Cursor != nil && p.Limiter.Limit.CursorDirection != nil
+}
+
+func (p *dslQueryProps) hasSequenceSort() bool {
+	for _, order := range p.Limiter.SortBy {
+		switch order.(type) {
+		case query.SortBySequence:
+			return true
+		default:
```

this is undefined behaviour which should probably throw an error